### PR TITLE
2024 Annual Report Hyperledger AnonCreds

### PR DESCRIPTION
**Pull request by swcurran**
Signed-off-by: Stephen Curran <swcurran@gmail.com>

### CODE REVIEW

**Review by petermetz — APPROVED** [review]

File: project-reports/2024/2024-annual-Hyperledger-AnonCreds.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+* Completion (pending a final editorial pass) of the AnonCreds v1 specification.
+* Support for AnonCreds verifiable credentials and verifiable presentations in W3C Verifiable Credentials Data Model Standard format, eliminating the longstanding complaint that AnonCreds VCs are “not compliant”.
+* Contribution of the AnonCreds v2 codebase, with new ZKP capabilities and pluggable support for several cryptographic signature schemes, including PS Signatures (implemented), BBS+ Signatures, and PS Post Quantum Signatures.
+
+### Next Year’s Goals
+
+_Guidance: What are the goals for the next year of the project? The goals should list what you want to achieve, not just what you know you can achieve. Feel free to include stretch goals and things that you are looking to explore in the next year. For example, are you working on major new features? Or are you concentrating on adoption, community growth, or documentation?_
+
+Recent Aries Working Group have been spent on defining a roadmap for 2024. The following is extracted from the discussions and notes from those meetings. The ordering is from activities well underway, to “reach for the clouds” ideas:
```
Comment (tkuhrt) [inline project-reports/2024/2024-annual-Hyperledger-AnonCreds.md]: Is this a typo? Or are all AnonCred discussions held in the Aries Working Group?

File: project-reports/2024/2024-annual-Hyperledger-AnonCreds.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+Hyperledger AnonCreds LFX Insights Page for [Calendar Year 2023](https://insights.lfx.linuxfoundation.org/foundation/hyp/overview?project=anoncreds&repository=all&dateFilters=2023-01-01%20to%202023-12-31&dateRange=2023-01-01%20to%202023-12-31&compare=PP&granularity=month&hideBots=true) (v. [2022](https://insights.lfx.linuxfoundation.org/foundation/hyp/overview?project=anoncreds&repository=all&dateFilters=2022-01-01%20to%202022-12-31&dateRange=2022-01-01%20to%202022-12-31&compare=PP&granularity=month&hideBots=true))  and for [Q1 2024 Oct-Dec.](https://insights.lfx.linuxfoundation.org/foundation/hyp/overview?project=anoncreds&repository=all&dateFilters=2023-10-01%20to%202023-12-31&dateRange=2023-10-01%20to%202023-12-31&compare=PP&granularity=month&hideBots=true) (v. [Q4 2023](https://insights.lfx.linuxfoundation.org/foundation/hyp/overview?project=anoncreds&repository=all&dateFilters=2023-07-01%20to%202023-09-30&dateRange=2023-07-01%20to%202023-09-30&compare=PP&granularity=month&hideBots=true)).
```
Comment (tkuhrt) [inline project-reports/2024/2024-annual-Hyperledger-AnonCreds.md:12]: From the insights page, it appears that most of the work is being done by @swcurran. What can be done to improve the activity by others in the community?

File: project-reports/2024/2024-annual-Hyperledger-AnonCreds.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+---
+layout: default
+title: 2024 Annual Review Hyperledger AnonCreds
+parent: 2024
+grand_parent: Project Updates
+---
+
+## Project Health
+
+_Guidance: Include a link to your project’s [LFX Insights page](https://insights-v2.lfx.linuxfoundation.org/projects). We will be looking for signs of consistent or increasing contribution activity. Please feel free to add commentary to add color to the numbers and graphs we will see on Insights._
```
Comment (denyeart) [inline project-reports/2024/2024-annual-Hyperledger-AnonCreds.md]: You can remove the instructions from the final report :-)

File: project-reports/2024/2024-annual-Hyperledger-AnonCreds.md (hunk 2 above)
Comment (swcurran) [reply project-reports/2024/2024-annual-Hyperledger-AnonCreds.md:12]: @ryjones -- can you check what repos are included in the AnonCreds LFX insights reports?  While I've done a lot with the spec, I've not done any coding and there has been a lot of code written this past year! I've done some of the code merging (after reviews by other devs), and perhaps that impacts the numbers?

File: project-reports/2024/2024-annual-Hyperledger-AnonCreds.md (hunk 3 above)
Comment (swcurran) [reply project-reports/2024/2024-annual-Hyperledger-AnonCreds.md]: I debated that, and thought they were useful to have for the casual reader, but I can remove them.

File: project-reports/2024/2024-annual-Hyperledger-AnonCreds.md (hunk 1 above)
Comment (swcurran) [reply project-reports/2024/2024-annual-Hyperledger-AnonCreds.md]: That's a typo -- sorry about that. I'll fix it.

Comment (swcurran) [reply project-reports/2024/2024-annual-Hyperledger-AnonCreds.md]: Fixed in next commit.

File: project-reports/2024/2024-annual-Hyperledger-AnonCreds.md (hunk 2 above)
Comment (swcurran) [reply project-reports/2024/2024-annual-Hyperledger-AnonCreds.md:12]: In the “Contributor Leaderboard”, I see me with 10% of the activities, followed by others at 8%, 5%, 3% and so on for at least 15 people with more than 1%.

I see the drop down with the list of repos. The only one I think is missing is anoncreds-v2-rs, so it is not that.

Love to have more participants 😄

File: project-reports/2024/2024-annual-Hyperledger-AnonCreds.md (hunk 3 above)
Comment (swcurran) [reply project-reports/2024/2024-annual-Hyperledger-AnonCreds.md]: Removed.

File: project-reports/2024/2024-annual-Hyperledger-AnonCreds.md (hunk 2 above)
Comment (ryjones) [reply project-reports/2024/2024-annual-Hyperledger-AnonCreds.md:12]: OK, I added the v2-rs and -tex repos

**Review by tkuhrt — APPROVED** [review]
Reviewed on the February 15th meeting.

# Hyperledger AnonCreds Annual Review Notes

## Project Health
* Project reports great strides in the past year
* Project reports lack of engagement in moving the AnonCreds v2.0 codebase forward as concerning
* LFX Insights seems to have inaccurate information
    * showing contributors from other projects that do not have anything to do with AnonCreds or the identity space
    * the same committer is showing up under different names (swcurran vs. Stephen Curran, as well as others)
    * organization mapping does not line up with contributor mapping
* Based on LFX Insights, it appears that the project contributions have decreased over the last 6 months of 2023
* Based on LFX Insights, it appears that new contributors are not staying
* Based on LFX Insights, the PR average wait time for first review is ~5 days and the average lead time (time it takes for pull requests to be raised, reviewed, accepted and merged) of ~9 days


## Maintainer Diversity
* 11 maintainers in the project representing 8 organizations
* This is great diversity for a project
* From what LFX Insights provides, it does appear that there is a fairly even distribution of activity across maintainers

## Project Adoption
* Hard to find information on — would there be projects using it as a dependency
* Getting organizations to contribute to and use AnonCreds v2 seems to be a challenge because of inaccurate perceptions with it being viewed as different to other VC standards and specification

## Goals
* 2023 Goals were not set due to new project status. A number of accomplishments did occur in 2023, including completion of a v1 specification and its implementation
* 2024 Goals - good set focused on formalizing the v1 specification into a standard, work towards v2 with a migration path between v1 and v2, and support for deploying W3C format.
* It might be worth mentioning in the goals work that will be done for fixing the perception of AnonCreds

## Help Required
* Need help from marketing the capabilities of AnonCreds
* Collaborating with others in the verifiable identity community on v2 with new signature schemes

## Lifecycle
* Project recommends remaining in incubation
* Concur with this recommendation

## Project Suggestions
* Blog posts, webinars, meetups and conference talks for AnonCreds could help with perception issues, even if driven by technologists 
* Working with the other communities within the LF via presenting to those groups could also help with perception issues
* Consider doing a 1.0 release to reflect the true status of the software

**Pull request merged**